thirdparty/faiss/faiss/IndexIVF.h

```diff
@@ -67,7 +67,9 @@ struct Level1Quantizer {
 struct IVFSearchParameters {
     size_t nprobe;    ///< number of probes at query time
     size_t max_codes; ///< max nb of codes to visit to do a query
-    IVFSearchParameters() : nprobe(1), max_codes(0) {}
+    int parallel_mode; // default value if -1, and we will use
+                       // this->parallel_mode in this case
+    IVFSearchParameters() : nprobe(1), max_codes(0), parallel_mode(-1) {}
```

Comment:
set parallel_mode default value to 0

Reply:
We cannot do that

We have this logic in `xxx_preassigned`:
```
    int preassigned_parallel_mode = 0;
    if (params && params->parallel_mode != -1) {
        preassigned_parallel_mode = params->parallel_mode;
    } else {
        preassigned_parallel_mode = this->parallel_mode;
    }
    int pmode = preassigned_parallel_mode & ~PARALLEL_MODE_NO_HEAP_INIT;
```
the old code is:
```
int pmode = this->parallel_mode & ~PARALLEL_MODE_NO_HEAP_INIT;
```
if default value is 0, assume some other codes pass the param in.
In the old perform: it will use `this->parallel_mode` as parallel_mode
In the new cold if we set default to 0: it will use 0 as parallel_mode. This is unsynced